src/test/scala/com/cognite/sdk/scala/common/DataPointsResourceBehaviors.scala

```diff
@@ -7,29 +7,61 @@ import org.scalatest.{FlatSpec, Matchers}
+  private val granularity = "1s"
+  private val aggregates = Seq("average", "stepInterpolation")
 
   def withTimeSeriesId(testCode: I => Any): Unit
 
-  def dataPointsResource(dataPoints: DataPointsResource[Id, I]): Unit = {
-    it should "be possible to insert and delete numerical data points" in withTimeSeriesId { timeSeriesId =>
-      dataPoints.insertById(timeSeriesId, testDataPoints)
+  def dataPointsResource(dataPoints: DataPointsResource[Id, I]): Unit =
+    it should "be possible to insert and delete numerical data points" in withTimeSeriesId {
+      timeSeriesId =>
+        dataPoints.insertById(timeSeriesId, testDataPoints)
 
-      Thread.sleep(3000)
-      val points = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
-      points should have size testDataPoints.size.toLong
+        Thread.sleep(3000)
+        val points = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
+        points should have size testDataPoints.size.toLong
 
-      val latest = dataPoints.getLatestDataPointById(timeSeriesId)
-      latest.isDefined should be (true)
-      val latestPoint = latest.get
-      testDataPoints.toList should contain (latestPoint)
+        val latest = dataPoints.getLatestDataPointById(timeSeriesId)
+        latest.isDefined should be(true)
+        val latestPoint = latest.get
+        testDataPoints.toList should contain(latestPoint)
 
-      dataPoints.deleteRangeById(timeSeriesId, startTime, endTime + 1)
-      Thread.sleep(10000)
-      val pointsAfterDelete = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
-      pointsAfterDelete should have size 0
+        val aggregatePointMap =
+          dataPoints.queryAggregatesById(
+            timeSeriesId,
+            startTime,
+            endTime,
+            granularity,
+            aggregates
+          )
+        aggregatePointMap("average") should have size testDataPoints.size.toLong
+        aggregatePointMap("stepInterpolation") should have size testDataPoints.size.toLong - 1
+        aggregatePointMap.keys should not contain ("sum")
+
+        dataPoints.deleteRangeById(timeSeriesId, startTime, endTime + 1)
+        Thread.sleep(10000)
+        val pointsAfterDelete = dataPoints.queryById(timeSeriesId, startTime, endTime + 1)
+        pointsAfterDelete should have size 0
+
+        val start: Long = 0
+        val aggregatePointExternalIdMap = dataPoints.queryAggregatesByExternalId(
+          "test__constant_74_with_noise",
+          start,
+          System.currentTimeMillis(),
+          "1d",
+          aggregates
+        )
+        assert(
+          aggregatePointExternalIdMap("average").size == aggregatePointExternalIdMap(
+            "stepInterpolation"
+          ).size + 1
+        )
+        assert(aggregatePointExternalIdMap.get("sum").isEmpty)
+        assert(aggregatePointExternalIdMap("average").head.value == 74.00018450606277)
```

Comment:
Use the `OptionValue` trait as described in http://www.scalatest.org/user_guide/using_OptionValues
```scala
trait DataPointsResourceBehaviors[I] extends Matchers with OptionValue
```